### DISCUSSION TITLE
`npm` protocol switch to `workspace` resolution

### DISCUSSION THREAD

**Discussion by articot** [yarnpkg/berry#4015] [Q&A]
I have monorepo with `libs` and `yunona` folders which is included in `workspaces` of root `package.json`

```json
  "workspaces": [    
    "libs/*"
    "yunona"
  ]
```

`libs` contain `private` and `public` libraries. `public` libraries are published to our private npm registry.

`yunona` is our web project built by `webpack` and references `private` `libs` via `workspace:` protocol and `public` `libs` via `npm:` protocol. 

```json
"dependencies": {
   "@bims/erarta": "npm:^1.13.1",
}
```

![Capture_1](https://user-images.githubusercontent.com/52530805/151003149-3f97f8b0-d45d-4c13-b689-7ec9487f6cad.PNG)

Now, our `public` library `erarta` has gone ahead and has version `1.14.1` (and published to npm), and I also have updated `package.json` of web project `yunona` with this this version, and now what I get after `yarn install`

![Capture_2](https://user-images.githubusercontent.com/52530805/151003873-05a60080-7e1d-4e6e-897d-8f41d6f19cb3.PNG)

which is quite unexpected, it now uses `workspace` protocol via `npm` ?

I expected it to still consume this package via npm registry - so I specifically set in package.json of web `yunona` project

```json
"dependencies": {
   "@bims/erarta": "npm:^1.14.1",
}
```

Can you explain this behavior, and how can I pin `npm:` protocol for good ?

thank you

**Comment by konsvarl** [comment]
Up!

**Comment by merceyz** [comment]
> Can you explain this behavior, and how can I pin npm: protocol for good ?

Even though you specify `npm:` Yarn will still match the range to your workspace, you'll need to disable `enableTransparentWorkspaces` if you only want explicit `workspace:` ranges to match your workspaces.

https://yarnpkg.com/configuration/yarnrc#enableTransparentWorkspaces
